### PR TITLE
Purescript 0.9.1 update

### LINKED ISSUE #29: 0.9.1 Compatibility
Compatibility with 0.9.1 appears to be blocked until the `purescript-datetime` rewrite is complete.

The only other issue I can see is changing `import Data.Maybe.Unsafe (fromJust)` to `import Data.Maybe (fromJust)`. I'll include that in a PR once `purescript-datetime` is ready. 👍

### PR DESCRIPTION
**Pull request by kika**
Please consider this PR for update to 0.9.1 compiler.
Fixes #29

### CODE REVIEW

**Comment by paulyoung** [comment]
I was about to do this myself.

File: bower.json [hunk 1]
```diff
@@ -15,17 +15,19 @@
...
+    "purescript-unsafe-coerce": "^1.0.0",
+    "purescript-nullable": "~1.0.0",
+    "purescript-node-streams": "~1.0.0",
+    "purescript-exceptions": "~1.0.0",
+    "purescript-js-date": "^1.0.0",
+    "purescript-globals": "^1.0.0"
   },
   "devDependencies": {
-    "purescript-console": "~0.1.1"
+    "purescript-console": "~1.0.0"
```
Comment (garyb) [inline bower.json]: Can you use `^` rather than `~` for all dependencies please?

**Comment by garyb** [comment]
@paf31 / @hdgarrood do you think we should be using `JSDate` or `DateTime` in this library? I'm torn...

File: src/Node/FS/Async.purs [hunk 2]
```diff
@@ -31,24 +31,28 @@ module Node.FS.Async
   , fdClose
   ) where
 
-import Prelude
-import Control.Monad.Eff
+import Prelude (Unit, bind, show, flip, ($), (<<<), (/), (<$>))
```
Comment (hdgarrood) [inline src/Node/FS/Async.purs]: Can you use an open import for Prelude please?

**Comment by hdgarrood** [comment]
Since it's easy to convert between the two, and given that quite a few people use this library directly, I'm tempted to say `DateTime`. We can always introduce new higher or lower level APIs later if it turns out they would be useful.

**Comment by garyb** [comment]
Ok, cool. That was my thinking on the pro-using-`DateTime` too.

File: bower.json (hunk 1 above)
Comment (kika) [inline bower.json]: Of course. I left the range specs as they were there previously.

Comment (garyb) [inline bower.json]: Yeah, it's a mixture of: bower used to default to `~`, and it didn't matter as we were on `0.x.y` anyway. As of the v1 releases using `^` means we can make additive changes to libraries without having to re-release things further down the chain.

**Comment by kika** [comment]
Import for Prelude opened (is this the current convention?), tilde ranges replaced with caret ones.

Not sure if there's work required on the JSData/DateTime: the low level stuff is in JSDate and all computations are in DateTime and friends. Seems very nice and logical to me.

**Comment by hdgarrood** [comment]
It is - the compiler lets you have one open import without warnings now, and so having Prelude as an open import means that you don't have to be updating the import list as often, and also reduces the risk of merge conflicts.

File: test/Streams.purs [hunk 3]
```diff
@@ -1,24 +1,20 @@
 module Test.Streams where
 
-import Prelude
-import Data.Maybe
-import Data.Either
+import Prelude (Unit, bind, (==))
```
Comment (hdgarrood) [inline test/Streams.purs]: If we make this open, can we get rid of the imports of Control.Apply and Control.Bind too?

**Comment by kika** [comment]
Done. Didn't know that compiler allows one open import. Very convenient.

@hdgarrood while I have your attention: in whose court is the `purescript-node-http` ball? I'm using my own copy (and own change which I did just to discover that @paf31  already submitted the PR), but all the requirements for @paf31's PR have been met as it seems.
And there's also `purescript-parsing` in a strange state: all changes made, but not version tagged.

**Comment by garyb** [comment]
`purescript-parsing` isn't quite done, there's a PR changing it to use `List` which is much more efficient. If I get a chance this afternoon I'll cherry-pick the commit and open a new PR for that change.

**Comment by kika** [comment]
@garyb I thought this PR make the build fail (though I believe it's just bad timing, infamous `-ffi` option) and thus was left aside.

**Comment by hdgarrood** [comment]
Looks great. Thanks very much!

**Comment by hdgarrood** [comment]
Re `node-http`, I think that PR looks fine, and the libraries we were waiting for have been updated, so I think now it just needs someone to update bower.json to point to real versions and tag it.

**Comment by paf31** [comment]
I can look at this later today unless someone beats me to it.

**Comment by paulyoung** [comment]
Will these changes make it into a release any time soon?

**Pull request merged**